tools/build_api.py

```diff
@@ -235,7 +235,7 @@ def build_project(src_path, build_path, target, toolchain_name,
 
             add_result_to_report(report, cur_result)
 
-        return res
+        return res, toolchain.map_outputs
```

Comment:
Do you think we could reuse add_result_to_report() or apply similar mechanism to collect the data, rather than change the return of build_project()?